sdk/java/pulumi/src/main/java/com/pulumi/provider/internal/infer/ComponentAnalyzer.java

```diff
@@ -68,25 +68,33 @@ public static PackageSpec generateSchema(Metadata metadata, Class<?>... classes)
 
         ComponentAnalyzer analyzer = new ComponentAnalyzer(metadata);
         Map<String, ResourceSpec> components = new HashMap<>();
-        
+
+        if (classes.length == 0) {
+            throw new IllegalArgumentException("At least one component class must be provided");
+        }
+
+        String namespace = classes[0].getPackage().getName().split("\\.")[1];
```

Comment:
Can there be no `.` at all?

Reply:
There can be. I think in that case we just leave the namespace empty.  I made it so now.